mmcv/utils/config.py

```diff
@@ -556,7 +556,7 @@ def __setstate__(self, state):
 
     def dump(self, file=None):
         cfg_dict = super(Config, self).__getattribute__('_cfg_dict').to_dict()
-        if self.filename.endswith('.py'):
+        if file is not None and file.endswith('.py'):
             if file is None:
                 return self.pretty_text
             else:
```

Comment:
Thanks for your contribution and the original dump logic is actually weird. The argument `file` is judged twice here, maybe the logic can be refined?